### PR TITLE
Add a function to update a track metadata

### PR DESCRIPTION
**Pull request by Guichaguri**
This PR adds the `updateMetadata(id, metadata)` function which allows you to update any track in the queue, including the current one. It returns a promise which resolves after the metadata has been successfully updated.

This PR supersedes #331 and implements #93. It is also the base requirement for #384.

#### Parameters
* `id`: The track id to update
* `metadata`: A subset of the track object containing only the `artwork`, `title`, `artist`, `album`, `description`, `genre`, `date`, `rating` and `duration` properties.

#### Rejections
As the function returns a promise, it can reject with the code `track_not_in_queue` when there's no track that matches the id.

### CODE REVIEW

**Comment by Guichaguri** [comment]
@dcvz can you take a look? The Android and Windows parts are already done.

**Comment by dcvz** [comment]
I'll take a look this week @Guichaguri 👍

**Comment by davichoso** [comment]
I tested it in android and its working fine

File: android/src/main/java/com/guichaguri/trackplayer/module/MusicModule.java [hunk 1]
```diff
@@ -241,6 +243,33 @@ public void remove(ReadableArray tracks, final Promise callback) {
         });
     }
 
+    @ReactMethod
+    public void updateMetadata(String id, ReadableMap map, final Promise callback) {
```
Comment (dcvz) [inline android/src/main/java/com/guichaguri/trackplayer/module/MusicModule.java]: let's rename this `updateMetadataForTrack`

Comment (Guichaguri) [reply android/src/main/java/com/guichaguri/trackplayer/module/MusicModule.java]: Sure

Comment (dcvz) [reply android/src/main/java/com/guichaguri/trackplayer/module/MusicModule.java]: I'm about to finish the iOS part so if you could rename them now, we can merge in a few min :)

File: android/src/main/java/com/guichaguri/trackplayer/module/MusicModule.java [hunk 2]
```diff
@@ -241,6 +243,33 @@ public void remove(ReadableArray tracks, final Promise callback) {
...
+                if(track.id.equals(id)) {
+                    index = i;
+                    break;
+                }
+            }
+
+            if(index == -1) {
+                callback.reject("track_not_in_queue", "No track found");
+            } else {
+                track.setMetadata(getReactApplicationContext(), Arguments.toBundle(map), binder.getRatingType());
```
Comment (dcvz) [inline android/src/main/java/com/guichaguri/trackplayer/module/MusicModule.java]: `Track` has a property called: `originalItem` could you please make sure that these properties get merged into the `originalItem` please. You can see that I did that in iOS.

Comment (Guichaguri) [reply android/src/main/java/com/guichaguri/trackplayer/module/MusicModule.java]: `setMetadata` already deals with that, see lines 112 to 113 from `Track.java`

File: docs/Documentation.md [hunk 3]
```diff
@@ -207,7 +207,8 @@ Gets the whole queue
 #### `removeUpcomingTracks()`
 Clears any upcoming tracks from the queue.
 
-#### `updateMetadata(id, metadata)`
+#### `updateMetadataForTrack(id, metadata)`
+Updates the metadata of a track in the queue.
```
Comment (dcvz) [inline docs/Documentation.md]: I would also add something like: `If the metadata of the current track is changed -- the Android Notification and the NowPlaying Center will be updated accordingly.

File: android/src/main/java/com/guichaguri/trackplayer/module/MusicModule.java (hunk 2 above)
Comment (dcvz) [reply android/src/main/java/com/guichaguri/trackplayer/module/MusicModule.java]: Oops! I must have missed that 😭

File: index.d.ts [hunk 4]
```diff
@@ -108,6 +111,7 @@ declare namespace RNTrackPlayer {
   export function skip(trackId: string): Promise<void>;
   export function skipToNext(): Promise<void>;
   export function skipToPrevious(): Promise<void>;
+  export function updateMetadata(id: string, metadata: TrackMetadata) : Promise<void>;
```
Comment (dcvz) [inline index.d.ts]: `updateMetadataForTrack`

**Comment by Guichaguri** [comment]
Awesome! Can I make a new release or are you already working on that right now?

**Comment by dcvz** [comment]
Go ahead :)

**Comment by dcvz** [comment]
for the changelog:

- I fixed a bug where if you reloaded RN, the player would continue playing

**Comment by dcvz** [comment]
@Guichaguri I just added a fix for #526

**Pull request merged**